automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala

```diff
@@ -56,18 +60,61 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
     val publishButton = Button("publish-button" inside this)
     val unpublishButton = Button("unpublish-button" inside this)
     val shareWorkspaceButton = Button("share-workspace-button" inside this)
+
+    def clickEdit(): Unit = {
+      editButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickSave(): Unit = {
+      saveButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickCancel(): Unit = {
+      cancelButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickClone(): Unit = {
+      cloneButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickDeleteWorkspace(): Unit = {
+      deleteWorkspaceButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickPublish(): Unit = {
+      publishButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickUnpublish(): Unit = {
+      unpublishButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickShareWorkspaceButton(): Unit = {
+      shareWorkspaceButton.doClick()
+      await condition new AclEditor().isOpen()
+    }
   }
 
   private val workspaceAttributesArea = Collapse("attribute-editor", new FireCloudView {
     override def awaitReady(): Unit = {
       table.awaitReady()
-      awaitState("enabled")
+      wsSummaryPage.awaitReady()
```

Comment:
I think when parent component is "ready" or "enabled", all sub components are "ready" or "enabled". If one sub-component is ready but while other sub-components are not, it could makes UI to move or jitter. negative side-effect chain reaction.

Reply:
Interesting point. However it makes sense for a page's ready to be implemented by delegating down to the sub component's ready. If the sub component invokes the parent ready then it'll be an endless loop. The checks have to be one way. I think it's then up to the previous action to know how wide the await needs to be... either just a sub component or the whole page.

Reply:
I still really dislike the parent references. Awaits should delegate down, not up.